assets/upload.js

```diff
@@ -59,18 +57,20 @@ window.addEventListener("load", function() {
     request.send(file);
   }
 
-  function recurse_upload(entry, base_url) {
+  function recurse_upload(entry, base_path) {
     if(entry.isFile) {
-      if(entry.file)
-        entry.file(function(f) {
-          upload_file(base_url + entry.fullPath, f);
-        });
-      else
-        upload_file(base_url + entry.fullPath, entry.getFile());
+      entry.file(function(f) {
+        let file_path = entry.fullPath.replace(/^\/+/, ""); // we don't want the leading '/'s
+        // each path segment needs to be individually encoded
+        // e.g. encodeURIComponent("some#/$path") == "some%23%2F%24path" which is wrong
+        //      we want "some%23/%24path" instead
+        file_path = file_path.split("/").map((seg) => encodeURIComponent(seg)).join("/");
+        upload_file(base_path + file_path, f);
+      });
     } else
-      entry.createReader().readEntries(function(e) {
-        e.forEach(function(f) {
-          recurse_upload(f, base_url)
+      entry.createReader().readEntries(function(e_arr) {
+        e_arr.forEach(function(f) {
+          recurse_upload(f, base_path)
         });
       });
```

Comment:
I'm pretty sure this original conditional logic is a screw-up.

Reply:
And I think it's correct because I wrote it that way. Some secrets cannot be known.